clang/test/SemaCXX/warn-dangling-local.cpp

```diff
@@ -34,6 +34,6 @@ struct basic_string {
 };
 }  // namespace std
 void test(const char* a) {
-  // verify we're emitting the `-Wdangling-assignment` warning.
+  // verify we're emitting the `-Wdangling-assignment-gsl` warning.
   a = std::basic_string().c_str(); // expected-warning {{object backing the pointer a will be destroyed at the end of the full-expression}}
```

Comment:
I'm a little confused here, since `a` is of type `const char*`, why do we mention the `-gsl` version of the warning here?